### PR TITLE
 feat: display snackbar with undo action on removing liked event

### LINKED ISSUE #1282: No message displayed when a liked event is removed
**Describe the feature you'd like**

When an event is removed from the list of liked events in the FavoritesFragment by clicking the heart icon on the particular event, a snackbar should be displayed notifying that the event has been removed from liked events.

**Screenshots**

Issue 1282

**Additional context**

Currently on removing the event, it just disappears.

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by addiegupta**
fixes Issue #1282 by displaying a snackbar message when a liked event is removed.
    The snackbar also has an Undo option which undos the remove operation and the event is displayed again in the RecyclerView

Changes: 
Display snackbar message that liked event has been removed
The snackbar has an undo action button which can undo the remove operation

Also fixed 2 grammatical errors in strings.xml

Screenshots for the change:

ezgif com-video-to-gif

### CODE REVIEW

**Comment by liveHarshit** [comment]
Why is it required?

**Comment by addiegupta** [comment]
Right now the event just disappears without any feedback that it has been removed so I thought this would be better. Also in the event that a user removes an event by mistake, thrn it can notify the user. And maybe an Undo option could be added later on

**Comment by addiegupta** [comment]
@liveHarshit  I have implemented the undo feature now

**Comment by addiegupta** [comment]
@liveHarshit can you review this? I have fixed the merge conflicts and implemented undo feature

**Comment by liveHarshit** [comment]
> Reset LGTM.

remove comments

**Comment by addiegupta** [comment]
The Codacy Bot raised a 'missing comments on public method' exception Ill remove them nevertheless

**Comment by addiegupta** [comment]
@liveHarshit please review this I have resolved conflicts

File: app/src/main/java/org/fossasia/openevent/general/favorite/FavoriteEventsRecyclerAdapter.kt [hunk 1]
```diff
@@ -9,10 +9,10 @@ import kotlinx.android.synthetic.main.item_card_favorite_event.view.favoriteFab
 import org.fossasia.openevent.general.R
 import org.fossasia.openevent.general.event.Event
 import org.fossasia.openevent.general.common.EventClickListener
-import org.fossasia.openevent.general.event.EventViewHolder
 import org.fossasia.openevent.general.common.EventsDiffCallback
 import org.fossasia.openevent.general.common.FavoriteFabClickListener
 import org.fossasia.openevent.general.common.ShareFabClickListener
+import org.fossasia.openevent.general.event.EventViewHolder
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/favorite/FavoriteEventsRecyclerAdapter.kt]: Why these changes?

File: app/src/main/java/org/fossasia/openevent/general/favorite/FavoriteFragment.kt [hunk 2]
```diff
@@ -133,9 +133,15 @@ class FavoriteFragment : Fragment() {
 
         val favFabClickListener: FavoriteFabClickListener = object : FavoriteFabClickListener {
             override fun onClick(event: Event, itemPosition: Int) {
-                favoriteEventViewModel.setFavorite(event.id, !event.favorite)
-                event.favorite = !event.favorite
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/favorite/FavoriteFragment.kt]: Why you remove this logic?

Comment (addiegupta) [reply app/src/main/java/org/fossasia/openevent/general/favorite/FavoriteFragment.kt]: I replaced `!event.favorite` with `false` as the statement always marks the favorite property as false. And the second line `event.favorite = !event.favorite` does not really achieve anything as the argument `event: Event` received is just a copy of the event . Please correct me if I'm wrong

File: app/src/main/java/org/fossasia/openevent/general/favorite/FavoriteEventsRecyclerAdapter.kt (hunk 1 above)
Comment (addiegupta) [reply app/src/main/java/org/fossasia/openevent/general/favorite/FavoriteEventsRecyclerAdapter.kt]: Made a mistake while resolving conflicts will resolve this

Comment (addiegupta) [reply app/src/main/java/org/fossasia/openevent/general/favorite/FavoriteEventsRecyclerAdapter.kt]: Resolved
